constellation/constellation.py

```diff
@@ -45,8 +45,8 @@ def status(self):
             x_status = x.status(self.prefix)
             print("    - {} ({}): {}".format(x.name, x_name, x_status))
 
-    def start(self, pull_images=False):
-        if any(self.containers.exists(self.prefix)):
+    def start(self, pull_images=False, subset=None):
+        if subset is None and any(self.containers.exists(self.prefix)):
```

Comment:
So we only check for existence if starting all containers? Don't we also want to check if containers in the subset already exist before trying to start them? I'm confused because I thought the idea here was to restart rather than just start a subset but I can't see any logic for stopping the containers first. Maybe I'm misunderstanding the intended usage...

Reply:
We *definitely* can't start with any (say) hint_ prefixed containers kicking around because either we're already running or we're in an inconsistent state. But if we are starting a subset we're probably putting ourselves into an incompletely-started case anyway. There's a case to be made for checking the subset containers that are going to be started, but that's already violated by the workers in hint